hbase-server/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java

```diff
@@ -247,6 +248,30 @@ public HFileSystem getHfs() {
     return this.hfs;
   }
 
+  /**
+   *
+   * @param clazz the class which need to check whether implements
+   *  org.apache.hadoop.fs.CanUnbuffer
+   * @return is implements CanUnbuffer true: Yes, false: No
+   */
+  @VisibleForTesting
+  public boolean isImplementsCanUnbuffer(Class<?> clazz){
+    Class<?>[] interfaces = clazz.getInterfaces();
+    if (interfaces.length != 0) {
+      for (Class<?> clz : interfaces) {
+        if (clz.getCanonicalName().toString().equals("org.apache.hadoop.fs.CanUnbuffer")) {
```

Comment:
HBase drops support of Hadoop 2.6/2.7. We should remove the use of reflection to detect this.
Use AnInterface.class.isAssignableFrom(anInstance.getClass()) instead?

Reply:
@jojochuang Thanks for the review! Try to do the modify tomorrow.

Reply:
@jojochuang Changed with AnInterface.class.isAssignableFrom(anInstance.getClass()) please review again. Thanks!